### PR TITLE
Setup

### LINKED ISSUE #2: Set up project
Make sure that all the following are working as expected:

- Typescript
- React
- Styled components
- Styled system (manages responsiveness)
- add custom theme using `ThemeProvider`

[Reading around styled component and styled system](https://rangle.io/blog/styled-components-styled-systems-and-how-they-work/#:~:text=Styled%20System%20is%20a%20collection,%2C%20colors%2C%20and%20layout%20properties.). Make sure you check out the use of ThemeProvider

### PR DESCRIPTION
**Pull request by raphaelj274**
### What changes have you made?

- Set up Typescript, Styled components, Styled system and added custom theme using ThemeProvider

### Which issue(s) does this PR fix?

- Fixes #2

### How to test

-  Edit theme.ts and check if website updates changes

### CODE REVIEW

File: next-env.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/// <reference types="next" />
```
Comment (Jbarget) [inline next-env.d.ts:1]: is this something thats needed by nextjs?

File: pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import Head from "next/head";
+import Image from "next/image";
+import styles from "../styles/Home.module.css";
```
Comment (Jbarget) [inline pages/index.tsx]: can delete this import too

File: pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import Head from "next/head";
+import Image from "next/image";
```
Comment (Jbarget) [inline pages/index.tsx]: can delete this import

**Review by Jbarget — CHANGES_REQUESTED** [review]
some minor last changes and a question

File: next-env.d.ts (hunk 1 above)
Comment (raphaelj274) [reply next-env.d.ts:1]: I think its needed for the tsc to pick up the nextjs types? So we can't remove it - At least thats what the docs say ahah

Comment (Jbarget) [reply next-env.d.ts:1]: nice nice, was just checking since i've never used nextjs before

**Review by Jbarget — APPROVED** [review]
nice work gang!

remember when you "squash and merge" to clean up the commit messages too in the text box it shows you 

image

**Comment by raphaelj274** [comment]
sets up Typescript, Styled components, Styled system and added custom theme using ThemeProvider #2

**Pull request merged**